internal/csi-common/nodeserver-default.go

```diff
@@ -112,6 +114,53 @@ func (ns *DefaultNodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.No
 
 	*/
 
+	stat, err := os.Stat(targetPath)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "failed to get stat for targetpath %q: %v", targetPath, err)
+	}
+
+	if stat.Mode().IsDir() {
+		return filesystemNodeGetVolumeStats(ctx, targetPath)
+	} else if (stat.Mode() & os.ModeDevice) == os.ModeDevice {
```

Comment:
`NodeGetVolumeStats` common for both cephfs and rbd do we need to return an error If the driver is cephfs?

Reply:
We can get the metrics for both CephFS and RBD. Maybe it is cleaner to split this out and have dedicated handlers for each node-plugin instead?

Reply:
Sounds reasonable to split it out.

Reply:
This is done now, please check again.